### ISSUE TITLE
clang-tidy `cppcoreguidelines-missing-std-forward` false positive on example from docs on Windows

### ISSUE THREAD

**Issue by pum1k** [llvm/llvm-project#209302]
The `cppcoreguidelines-missing-std-forward` clang-tidy triggers a warning on the example from the docs, where it should be fine.

Reproducer:
```c++
#include <utility>

template <typename T>
void impl(T);

template <class T>
void wrapper(T &&t)
{
    impl(std::forward<T>(t), 1, 2); // Correct
}
```

Expected result: No warning produced
Actual result:
```
> clang-tidy --checks=-*,cppcoreguidelines-missing-std-forward test.cpp --
1 warning generated.
D:\projects\CVUT-FEL\bp\experimental\vision-sim3d\test.cpp:7:18: warning: forwarding reference parameter 't' is never forwarded inside the function body [cppcoreguidelines-missing-std-forward]
    7 | void wrapper(T &&t)
      |                  ^
```

This issue occured on Windows on both 20.1.0 and the current version. I was not able to reproduce this issue on other platforms. Furthermore, when using clangd in editor, it handles the check correctly.

```
> clang-tidy --version                                                    
LLVM (http://llvm.org/):
  LLVM version 22.1.0
  Optimized build.
```

**Comment by vbvictor** [comment]
For windows to work you should pass `-fno-delayed-template-parsing`, see
https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/test/clang-tidy/checkers/cppcoreguidelines/missing-std-forward.cpp#L1

**Comment by vbvictor** [comment]
I'm not into clangd internals, but it may work because it forces all templates to be parsed (I imagine for code navigation and so on), so it implicitly does "no-delayed-template-parsing"

**Comment by pum1k** [comment]
> For windows to work you should pass `-fno-delayed-template-parsing`, see https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/test/clang-tidy/checkers/cppcoreguidelines/missing-std-forward.cpp#L1

Thanks, this works.
Also, it looks like it works by default when using std >= c++20.
